src/devices/DCMotor/DCMotor2Pin.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device.DCMotor
+{
+    internal class DCMotor2Pin : DCMotor
+    {
+        private PwmChannel _pwm;
+        private int _pin1;
+        private double _speed;
+
+        public DCMotor2Pin(
+            PwmChannel pwmChannel,
+            int pin1,
+            GpioController controller,
+            bool shouldDispose)
+            : base(controller ?? ((pin1 == -1) ? null : new GpioController()), controller == null ? true : shouldDispose)
+        {
+            _pwm = pwmChannel;
+
+            _pin1 = pin1;
+
+            _speed = 0;
+
+            _pwm.Start();
+
+            if (_pin1 != -1)
+            {
+                Controller.OpenPin(_pin1, PinMode.Output);
+                Controller.Write(_pin1, PinValue.Low);
+            }
+        }
+
+        public override void Start()
+        {
+            _pwm.Start();
+        }
+
+        public override void Stop()
+        {
+            _pwm.Stop();
+        }
+
+        /// <summary>
+        /// Gets or sets the speed of the motor.
+        /// Speed is a value from -1 to 1
+        /// 1 means maximum speed, signed value changes the direction.
+        /// </summary>
+        public override double Speed
+        {
+            get
+            {
+                return _speed;
+            }
+            set
+            {
+                double val = Math.Clamp(value, _pin1 != -1 ? -1.0 : 0.0, 1.0);
+
+                if (_speed == val)
+                {
+                    return;
+                }
+
+                if (val >= 0.0)
+                {
+                    if (_pin1 != -1)
+                    {
+                        Controller.Write(_pin1, PinValue.High);
+                    }
+
+                    _pwm.DutyCycle = val;
+                }
+                else
+                {
+                    if (_pin1 != -1)
+                    {
+                        Controller.Write(_pin1, PinValue.Low);
+                    }
+
+                    _pwm.DutyCycle = Math.Abs(val);
```

Comment:
why do you Abs if you know that value is negative and is `>= -1`

Reply:
probably because i saw it in DCMotor3Pin 😄